docs/user_guide.md

```diff
@@ -1326,6 +1331,97 @@ common arguments:
                         Set verbosity level.
 ~~~~~~~~~~~~~~~~~~~~~
 
+### <a name="source-components"></a> Source components (`components`)
+
+~~~~~~~~~~~~~~~~~~~~~
+usage: CodeChecker cmd components [-h] [--url PRODUCT_URL]
+                                  [--verbose {info,debug,debug_analyzer}]
+                                  {list,add,del} ...
+
+Source components are named collection of directories specified as directory
+filter.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+available actions:
+  {list,add,del}
+    list                List source components available on the server.
+    add                 Creates a new source component.
+    del                 Delete a source component from the server.
+~~~~~~~~~~~~~~~~~~~~~
+
+
+#### <a name="new-source-components"></a> New/Edit source component
+
+~~~~~~~~~~~~~~~~~~~~~
+usage: CodeChecker cmd components add [-h] [--description DESCRIPTION] -i
+                                      COMPONENT_FILE [--url PRODUCT_URL]
+                                      [--verbose {info,debug,debug_analyzer}]
+                                      NAME
+
+Creates a new source component or updates an existing one.
+
+positional arguments:
+  NAME                  Unique name of the source component.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --description DESCRIPTION
+                        A custom textual description to be shown alongside the
+                        source component.
+  -i COMPONENT_FILE, --import COMPONENT_FILE
+                        Path to the source component file. Please consult the
```

Comment:
I would put a simple example here and a not that for further details check the user guide.